test/libsolidity/syntaxTests/array/concat/bytes_concat_wrong_type_zero_literal.sol

```diff
@@ -0,0 +1,31 @@
+contract C {
+    function f() public pure {
+        bytes.concat(
+            0,
+            -0,
+            0.0,
+            -0.0,
+            0e10,
+            -0e10,
+            0e-10,
+            -0e-10,
+            (0),
+            0x00,
+            -0x00,
+            -0x0000000000000000000000000000000000000000000000000000000000000000
```

Comment:
It's interesting that `-0x0000000000000000000000000000000000000000000000000000000000000000` is not convertible to `bytes32` while `0x0000000000000000000000000000000000000000000000000000000000000000` is but makes sense for consistency with other negative hex numbers.